### DISCUSSION TITLE
Permission denied when accessing media devices (iOS)

### DISCUSSION THREAD

**Discussion by bergutman** [open-webui/open-webui#3154] [General]
# Bug Report

## Description

**Bug Summary:**
The voice functionality in OWUI does not work on Safari for iOS. The page never requests microphone permissions and returns the error message "Permission denied when accessing media devices" when attempting to use STT or the hands-free call mode. I'm not 100% sure if this issue is being caused by the fact that I am attempting to access OWUI via a local IP address without HTTPS. 

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

1. I ran OWUI with the following: docker run -d -p 3000:8080 --gpus=all -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama

2. I connected to OWUI from my phone at 172.16.101.155:3000 (my internal IP)

3. I attempted to launch the hands-free call mode and received this error message.

NOTE: The voice mode is working fine from other machines
 
**Expected Behavior:**
The page should request microphone permissions and then launch the call mode.

**Actual Behavior:**
Red error message "Permission denied when accessing media devices"

## Environment

- **Open WebUI Version:** latest bundled OWUI+Ollama docker image
- **Ollama (if applicable):** Using OpenAI API

- **Operating System:** Client: iOS Server: Gentoo
- **Browser (if applicable):** Safari iOS

## Reproduction Details
See above steps.
**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
![image](https://github.com/open-webui/open-webui/assets/2351549/71e83bb3-5fd3-4100-a4f7-daf49796915e)

## Installation Method

Docker Ollama bundle CPU only but tested GPT-4o API.

## Additional Information

I'm using local whisper for STT and OpenAI for the TTS.

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
Yep, needs to be served over HTTPS 😂

**Reply by bergutman** [reply]
Whelp... I played myself 💀

**Reply by gelbzucht** [reply]
@bergutman I played myself on Chrome desktop in the same way. ;) 
Besides switching to HTTPS, i was curious and got it working when setting an exception in Chrome for testing. 
Maybe it can be done in Safari in a similar way. 
`chrome://flags/#unsafely-treat-insecure-origin-as-secure`

via https://stackoverflow.com/questions/52759992/how-to-access-camera-and-microphone-in-chrome-without-https
#teamHTTPS ;)
